tests/test_cases/test_vhdl_zerovector/test_vhdl_zerovector.py

```diff
@@ -0,0 +1,49 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import cocotb
+from cocotb.result import TestFailure
+from cocotb.triggers import Timer
+
+
+@cocotb.test()
+async def test_long_signal(dut):
+    """ Write and read a normal signal (longer than 0)."""
+    await Timer(1, "ns")
+    dut.data_in <= 0x5
+    await Timer(1, "ns")
+
+    if dut.data_out != 0x5:
+        raise TestFailure("Failed to readback dut.data_out")
+
+
+@cocotb.test(expect_error=AttributeError if cocotb.SIM_NAME.lower().startswith(("ghdl", "xmsim", "ncsim")) else ())
+async def test_read_zero_signal(dut):
+    """ Read a zero vector. It should always read 0."""
+    await Timer(1, "ns")
+    if dut.Cntrl_out != 0:
+        raise TestFailure("Failed to readback dut.Cntrl_out")
+
+
+@cocotb.test(expect_error=AttributeError if cocotb.SIM_NAME.lower().startswith(("ghdl", "xmsim", "ncsim")) else ())
+async def test_write_zero_signal_with_0(dut):
+    """ Write a zero vector with 0."""
+    await Timer(1, "ns")
+    dut.Cntrl_out <= 0x0
+    await Timer(1, "ns")
+    if dut.Cntrl_out != 0:
+        raise TestFailure("Failed to readback dut.Cntrl_out")
+
+
+@cocotb.test(expect_error=AttributeError if cocotb.SIM_NAME.lower().startswith(("ghdl", "xmsim", "ncsim")) else ())
+async def test_write_zero_signal_with_1(dut):
+    """ Write a zero vector with 1. Should catch a "out of range" exception."""
+    await Timer(1, "ns")
+
+    try:
+        dut.Cntrl_out <= 0x1
+    except OverflowError:
+        pass
+    else:
+        assert False, "Exception did not occur"
```

Comment:
```suggestion
with pytest.raises(OverflowError):
    dut.Cntrl_out <= 0x1
```
You'll also need to import `pytest` at the top.

Reply:
This is probably a different topic:
Why do I get an "OverflowError" here, but a "TruncationError" if I do the same with BinaryValue class?

Reply:
`TruncationError`? I haven't seen that one. But they are two different pieces of code doing the checks and there was probably no attempt to keep them consistent. That should probably be changed. A different PR of course =)

Reply:
in tests/pytest/test_binary_value.py I had to supress with 
```
with pytest.warns(RuntimeWarning, match=TRUNCATION_MATCH):
     bin4 <= 5
```
Definitely a different PR